src/cascadia/TerminalApp/lib/ParentPane.cpp

```diff
@@ -0,0 +1,777 @@
+//   available space, and the percent of the space they respectively consume,
+//   which is stored in _desiredSplitPosition
+// Arguments:
+// - rootSize: The dimensions in pixels that this pane (and its children should consume.)
+// Return Value:
+// - <none>
+void ParentPane::_CreateRowColDefinitions(const Size& rootSize)
+{
+    if (_splitState == SplitState::Vertical)
+    {
+        _root.ColumnDefinitions().Clear();
+
+        // Create two columns in this grid: one for each pane
+        const auto paneSizes = _CalcChildrenSizes(rootSize.Width);
+
+        auto firstColDef = Controls::ColumnDefinition();
+        firstColDef.Width(GridLengthHelper::FromPixels(paneSizes.first));
+
+        auto secondColDef = Controls::ColumnDefinition();
+        secondColDef.Width(GridLengthHelper::FromPixels(paneSizes.second));
+
+        _root.ColumnDefinitions().Append(firstColDef);
+        _root.ColumnDefinitions().Append(secondColDef);
+    }
+    else
+    {
+        _root.RowDefinitions().Clear();
+
+        // Create two rows in this grid: one for each pane
+        const auto paneSizes = _CalcChildrenSizes(rootSize.Height);
+
+        auto firstRowDef = Controls::RowDefinition();
+        firstRowDef.Height(GridLengthHelper::FromPixels(paneSizes.first));
+
+        auto secondRowDef = Controls::RowDefinition();
+        secondRowDef.Height(GridLengthHelper::FromPixels(paneSizes.second));
+
+        _root.RowDefinitions().Append(firstRowDef);
+        _root.RowDefinitions().Append(secondRowDef);
+    }
+}
+
+// Method Description:
+// - Called after ctor to start listening to our children and append their xaml 
+//   elements. Call this only when both children can be attached (ep. they are not 
+//   already attached somewhere).
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void ParentPane::InitializeChildren()
+{
+    _root.Children().Append(_firstChild->GetRootElement());
+    _root.Children().Append(_secondChild->GetRootElement());
+    _SetupChildEventHandlers(true);
+    _SetupChildEventHandlers(false);
+}
+
+// Method Description:
+// - Setups all the event handlers we care about for a given child.
+// - It is called on initialization, for both children, and when one of our children
+//   changes (this is, it gets splitted or collapsed after split).
+// Arguments:
+// - firstChild - true to deal with first child, false for second child
+// Return Value:
+// - <none>
+void ParentPane::_SetupChildEventHandlers(bool firstChild)
+{
+    auto& child = firstChild ? _firstChild : _secondChild;
+    auto& closedToken = firstChild ? _firstClosedToken : _secondClosedToken;
+    auto& typeChangedToken = firstChild ? _firstTypeChangedToken : _secondTypeChangedToken;
+
+    if (const auto childAsLeaf = std::dynamic_pointer_cast<LeafPane>(child))
+    {
+        // When our child is a leaf and got closed, we, well, just close him.
+        closedToken = childAsLeaf->Closed([=, &child](auto&& /*s*/, auto&& /*e*/) {

```

Comment:
This is operating on a copy, isn't it? (and below, see my other comment about needing a dose of the weak/strong stuff).